File: Periodo_01/lab_icc1/baralho.c
#include <stdio.h>

const char naipes[] = {'C', 'O', 'P', 'E'};
const char valores[] = {'A', '2', '3', '4', '5', '6', '7', '8', '9', '1', 'J', 'Q', 'K'};

int main() {
    int n;
    scanf("%d", &n);

    char valor, naipe;
    
    char carta[3];

    char baralho[4][13];

    for(int x = 0; x < n; x++) {
        scanf("%s", &carta);

        if(carta[2] == 0) {
            valor = carta[0];
            naipe = carta[1];
        } else {
            valor = '1';
            naipe = carta[2];
        }
        for(int i = 0; i < 4; i++) {
            for(int j = 0; j < 13; j++) {
                if (valor == valores[j]);
            }
        }
        printf("%s %c\n", valor, naipe);
    }



    // for(int i = 0; i < 4; i++) {
    //     for(int j = 0; j < 13; j++) {
            
    //     }
    // }
}

File: Periodo_03/object_oriented/Java/VideoPoker/Baralho.java
import java.util.Collections;
import java.util.Random;
import java.util.Vector;

public class Baralho {
    private Vector<Carta> baralho = new Vector<Carta>();
    private Random rand;

    // Gera as 52 cartas do baralho (naipes em ordem reversa)
    public Baralho(int seed) {
        for(int naipe = 1; naipe < 5; naipe++) {
            for(int valor = 2; valor < 15; valor++) {
                baralho.add(new Carta(valor, naipe));
            }
        }

        // Inicializa rand de acordo com a seed
        if (seed == 0)
            rand = new Random();
        else
            rand = new Random(seed);

        // Embaralha as 52 cartas 
        baralhoEmbaralhar();
    }

    // Embaralha
    public void baralhoEmbaralhar() {
        Collections.shuffle(baralho, rand);
    }

    public Vector<Carta> getBaralho() {
        return baralho;
    }

    public void recomporBaralho(Mesa mesa) {
        Carta[] mao = mesa.getMao();
        for(Carta c: mao) {
            baralho.add(c);
        }
        Vector<Carta> descartadas = mesa.getDescartadas();
        for(Carta c: descartadas) {
            baralho.add(c);
        }
    }
}
